### PR TITLE
Supporting knowledgebases

### PR DESCRIPTION
**Pull request by tosan88**

### CODE REVIEW

File: tmereader/repository.go [hunk 1]
```diff
@@ -7,6 +7,23 @@ import (
...
+var sourceName = [...]string{
+	"authorityfiles",
+	"knowledgebases",
+}
+
+var sourcePathSuffix = [...]string{
+	"terms",
+	"eng/categories",
+}
+
```
Comment (BerniVarga) [inline tmereader/repository.go:33]: Couldn't we group all the knowledge base related info in one struct (for example), and the authority file related info in another structure? For me this way of defining enums seems a little bit complicated to understand and follow...

File: tmereader/repository.go [hunk 2]
```diff
@@ -80,7 +98,8 @@ func (t *tmeRepository) GetTmeTermsFromIndex(startRecord int) ([]interface{}, er
 }
 
 func (t *tmeRepository) getTmeTermsInChunks(startPosition int, maxRecords int) ([]interface{}, error) {
-	req, err := http.NewRequest("GET", fmt.Sprintf("%s/rs/authorityfiles/%s/terms?maximumRecords=%d&startRecord=%d", t.tmeBaseURL, t.taxonomyName, maxRecords, startPosition), nil)
+	url := fmt.Sprintf("%s/rs/%s/%s/%s?maximumRecords=%d&startRecord=%d", t.tmeBaseURL, t.source.String(), t.taxonomyName, sourcePathSuffix[t.source], maxRecords, startPosition)
```
Comment (BerniVarga) [inline tmereader/repository.go]: If you take the source through a String method, i would do something similar for the path suffix too (PathSuffix()). Or we can take the string simply from the array, without a function, keeping both of the references consistent.

File: tmereader/repository_test.go [hunk 3]
```diff
@@ -139,7 +139,7 @@ func (*dummyTransformer) UnMarshallTerm(content []byte) (interface{}, error) {
 }
 
 func repo(c dummyClient) Repository {
-	return &tmeRepository{httpClient: &c, tmeBaseURL: c.tmeBaseURL, accessConfig: tmeAccessConfig{userName: "test", password: "test", token: "test"}, maxRecords: 100, slices: 1, taxonomyName: "GL", transformer: new(dummyTransformer)}
+	return &tmeRepository{httpClient: &c, tmeBaseURL: c.tmeBaseURL, accessConfig: tmeAccessConfig{userName: "test", password: "test", token: "test"}, maxRecords: 100, slices: 1, taxonomyName: "GL", source:AuthorityFiles, transformer: new(dummyTransformer)}
```
Comment (BerniVarga) [inline tmereader/repository_test.go]: Please add tests for knowledgeBase option. Also, maybe rename this test, or the repo, for something indicating the type of the Repository that will be used.

**Comment by catalin-ilea** [comment]
+1

**Comment by BerniVarga** [comment]
+1

**Pull request merged**